sway/commands/input.c

```diff
@@ -86,6 +87,15 @@ struct cmd_results *cmd_input(int argc, char **argv) {
 	}
 
 	if (!res || res->status == CMD_SUCCESS) {
+		char *error = NULL;
+		struct xkb_keymap *keymap = sway_keyboard_compile_keymap(
+				config->handler_context.input_config, &error);
+		if (!keymap) {
+			return cmd_results_new(CMD_FAILURE, "Failed to compile keymap: %s",
+					error ? error : "(null)");
```

Comment:
Is not 'error' leaking here?
Maybe it's better to log errors when they appear?

Reply:
> Is not 'error' leaking here?

Fixed

> Maybe it's better to log errors when they appear?

The errors are being logged when they appear. There is to provide a quick description in swaymsg/swaynag  for user convenience.

Reply:
Also it's better to print something more descriptive instead of '(null)', very confusing

Reply:
I'll change it to `(error allocation failed)` when I get back to my laptop

Reply:
Maybe it's not allocation error, maybe callback was not called?
Or on any error it will be called?

Reply:
I used `(details unavailable)`, which isn't really helpful either, but shouldn't be confusing. The two causes would be the allocation failed or xkbcommon just didn't log anything (for whatever reason).